[BUGFIX] Header files import concat

stefanpenner opened the pull request:
This ensures concats are stable, and predictably ordered. This also fixes a prepend bug for test css.

stefanpenner commented:
will fix patch, just a sec.

homu commented:
:umbrella: The latest upstream changes (presumably bb4b51b) made this pull request unmergeable. Please resolve the merge conflicts.

stefanpenner commented:
rebased + retargeted to master.

stefanpenner commented:
Added tests: https://github.com/ember-cli/ember-cli/pull/6081/commits/4528501640c187444e2445bd6cf16cf73bfd2656#diff-1cef190fe22e6c889bb23fda5a8296a3R956

Release: test styles prepend: true, wasn't working and also fixed that.

stefanpenner commented:
windows test failure seems isolated to:

```
AssertionError: expected { Object (headerFiles, outputFile, ...) } to deeply equal { Object (annotation, headerFiles, ...) }
    at Context.<anonymous> (C:\projects\ember-cli\tests\unit\broccoli\ember-app-test.js:1044:33)
    at callFn (C:\projects\ember-cli\node_modules\mocha\lib\runnable.js:334:21)
    at Test.Runnable.run (C:\projects\ember-cli\node_modules\mocha\lib\runnable.js:327:7)
    at Runner.runTest (C:\projects\ember-cli\node_modules\mocha\lib\runner.js:429:10)
    at C:\projects\ember-cli\node_modules\mocha\lib\runner.js:535:12
    at next (C:\projects\ember-cli\node_modules\mocha\lib\runner.js:349:14)
    at C:\projects\ember-cli\node_modules\mocha\lib\runner.js:359:7
    at next (C:\projects\ember-cli\node_modules\mocha\lib\runner.js:285:14)
    at C:\projects\ember-cli\node_modules\mocha\lib\runner.js:322:7
    at done (C:\projects\ember-cli\node_modules\mocha\lib\runnable.js:291:5)
```

stefanpenner commented:
@ember-cli/core this should be ready for a review

cc @hjdivad

Turbo87 commented on tests/unit/broccoli/ember-app-test.js:
```diff
@@ -938,20 +939,188 @@ describe('broccoli/ember-app', function() {
+            "vendor/ember-cli/vendor-suffix.js"
+          ],
+          outputFile: "/assets/vendor.js",
+          separator: EOL + ';',
```
I'm a bit surprised by this. I though we used `\n` on Windows too?

stefanpenner replied:
@Turbo87 we actually explicitly specify EOL [here](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L1272). But we may want to investigate this as it doesn't seem right. As this merely tests existing behavior, should we consider the change as part of this PR or a subsequent. I can be convinced of either

Turbo87 replied:
let's investigate and fix in a different PR. I was just wondering where that came from.

stefanpenner commented:
landing this in canary.
